packages/sdk-demo/src/apps/SDKDemo/SDKDemo.tsx

```diff
@@ -78,45 +87,71 @@ const SDKDemoComponent = ({
     },
   });
 
-  const fetchToken = () =>
-    authData
-      ? fetchTokenBase(apiUrl, authData).catch(logout)
-      : Promise.reject();
+  const [fetchTokenError, setFetchTokenError] = useState<Error | null>(null);
+  const fetchToken = useCallback(async () => {
+    try {
+      if (!authData) throw new Error('Auth data is not provided');
+      if (authData instanceof Error) throw authData;
+      const token = await fetchTokenBase(apiUrl, authData);
+      setFetchTokenError(null);
+      return token;
+    } catch (error) {
+      setFetchTokenError(
+        error instanceof Error ? error : new Error(JSON.stringify(error))
```

Comment:
Instead of using `JSON.stringify(error)`, which may not always produce a meaningful message, consider using `String(error)` to convert the error to a string

Reply:
`String(error)` returns `'[object Object]'` while `JSON.stringify` returns a verbose output.

Reply:
Actually, String(error) only returns "[object Object]" if error is a plain object, not an actual Error instance. If it’s an Error object, it should return something like "Error: Some error". When you call String() on error message .toString is invoked

JSON.stringify(error) does give a more detailed output, but it might miss some properties of Error objects, like message or stack, unless those are explicitly enumerable.

Reply:
1. It's a demo project
2. error handling is tested in real cases